### PR TITLE
Addition of the getApiDisplayValue method

### PR DESCRIPTION
**Pull request by arek1-618**
Addition of the getApiDisplayValue method

### CODE REVIEW

File: modules/Vtiger/uitypes/MultiImage.php [hunk 1]
```diff
@@ -232,11 +250,11 @@ public function getListViewDisplayValue($value, $record = false, $recordModel =
 				return '';
 			}
 			$len = count($value);
-			for ($i = 0; $i < $len; $i++) {
+			for ($i = 0; $i < $len; ++$i) {
 				$val = $value[$i];
 				$result .= $val['name'] . ', ';
 			}
-			return \App\Purifier::encodeHtml(trim($result, "\n\s\t ,"));
+			return \App\Purifier::encodeHtml(trim($result, "\n\\s\t ,"));
```
Comment (TomaszQr) [inline modules/Vtiger/uitypes/MultiImage.php]: ?

File: modules/Vtiger/uitypes/MultiImage.php [hunk 2]
```diff
@@ -192,26 +193,43 @@ public function getDisplayValue($value, $record = false, $recordModel = false, $
 			if (!is_array($value)) {
 				return '';
 			}
-			for ($i = 0; $i < $len; $i++) {
+			for ($i = 0; $i < $len; ++$i) {
 				$val = $value[$i];
 				$result .= $val['name'] . ', ';
 			}
-			return \App\Purifier::encodeHtml(trim($result, "\n\s\t ,"));
+			return \App\Purifier::encodeHtml(trim($result, "\n\\s\t ,"));
```
Comment (TomaszQr) [inline modules/Vtiger/uitypes/MultiImage.php]: ?

File: modules/Vtiger/uitypes/MultiImage.php [hunk 3]
```diff
@@ -138,7 +139,7 @@ public function getHistoryDisplayValue($value, Vtiger_Record_Model $recordModel)
 			return $v['name'];
 		}, $value);
 		$result = implode(', ', $value);
-		return trim($result, "\n\s\t, ");
+		return trim($result, "\n\\s\t, ");
```
Comment (TomaszQr) [inline modules/Vtiger/uitypes/MultiImage.php]: ?

File: api/webservice/Core/Modules/Vtiger/uitypes/MultiImage.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+ * UIType MultiImage Field Class.
+ *
+ * @package   Api
+ *
+ * @copyright YetiForce Sp. z o.o
+ * @license   YetiForce Public License 3.0 (licenses/LicenseEN.txt or yetiforce.com)
+ * @author    Arkadiusz Adach <a.adach@yetiforce.com>
+ */
+
+namespace Api\Core\Modules\Vtiger\uitypes;
```
Comment (rskrzypczak) [inline api/webservice/Core/Modules/Vtiger/uitypes/MultiImage.php]:
```suggestion
namespace Api\Core\Modules\Vtiger\UiTypes;
```

File: api/webservice/Portal/BaseModule/RecordsList.php [hunk 5]
```diff
@@ -21,6 +21,7 @@ class RecordsList extends \Api\Core\BaseAction
 	 */
 	public function get()
 	{
+		\Vtiger_Field_Model::setDefaultUiTypeClassName('\\Api\\Core\\Modules\\Vtiger\\uitypes\\Base');
```
Comment (rskrzypczak) [inline api/webservice/Portal/BaseModule/RecordsList.php]:
```suggestion
		\Vtiger_Field_Model::setDefaultUiTypeClassName('\\Api\\Core\\Modules\\Vtiger\\UiTypes\\Base');
```

File: api/webservice/Core/Modules/Vtiger/uitypes/Base.php [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+{
+	/**
+	 * {@inheritdoc}
+	 */
+	public static function getInstanceFromField($fieldModel)
+	{
+		$uiTypeClassSuffix = ucfirst($fieldModel->getFieldDataType());
+		$instance = false;
+		foreach ([$fieldModel->getModuleName(), 'Vtiger'] as $moduleName) {
+			$className = "\\Api\\Core\\Modules\\{$moduleName}\\uitypes\\{$uiTypeClassSuffix}";
```
Comment (rskrzypczak) [inline api/webservice/Core/Modules/Vtiger/uitypes/Base.php]:
```suggestion
			$className = "\\Api\\Core\\Modules\\{$moduleName}\\UiTypes\\{$uiTypeClassSuffix}";
```

**Pull request merged**